src/foam/mlang/expr/DateGroupingExpr.js

```diff
@@ -81,14 +81,14 @@ foam.CLASS({
     {
       name: 'compare',
       code: function(o1, o2) {
-        let a1 = this.f(o1);
-        let a2 = this.f(o2);
-        return foam.String.compare(a1, a2);
+        let a1 = new Date(o1?.created);
+        let a2 = new Date(o2?.created);
+        return foam.Date.compare(a1, a2);
```

Comment:
This is being hard-coded to 'created' instead of using 'f'?  Do you need to create new Date()s if you just call foam.compare()?

Reply:
f is also hardcoded to created, seems like that is what we use this for? not really sure. Also foam.compare isnt a function in js

Reply:
I mean foam.util.compare()